src/solc_0.5/Land/erc721/LandBaseTokenV3.sol

```diff
@@ -154,12 +240,72 @@ contract LandBaseTokenV3 is ERC721BaseTokenV2 {
         _checkBatchReceiverAcceptQuad(msg.sender, address(0), to, size, x, y, data);
     }
 
-    function _idInPath(uint256 i, uint256 size, uint256 x, uint256 y) internal pure returns(uint256) {
+    function checkAndClearOwner(
+        uint256 size,
+        uint256 x,
+        uint256 y,
+        Land[] memory mintedLand,
+        uint256 index,
+        uint256 quadCompareSize
+    ) internal {
+        (uint256 layer, , ) = _getQuadLayer(quadCompareSize);
+        uint256 toX = x + size;
+        uint256 toY = y + size;
+
+        if (size >= quadCompareSize) {
+            for (uint256 xi = x; xi < toX; xi += quadCompareSize) {
+                for (uint256 yi = y; yi < toY; yi += quadCompareSize) {
+                    bool isQuadChecked = isQuadMinted(mintedLand, xi, yi, quadCompareSize, index);
+                    if (!isQuadChecked) {
+                        uint256 id = _getQuadId(layer, xi, yi);
+                        address owner = address(uint160(_owners[id]));
+
+                        if (owner == msg.sender) {
+                            mintedLand[index] = Land({x: xi, y: yi, size: quadCompareSize});
+                            index++;
+                            _owners[id] = 0;
+                        } else {
+                            require(owner == address(0), "Already minted");
+                        }
+                    }
+                }
+            }
+        }
+
+        quadCompareSize = quadCompareSize / 2;
+        if (quadCompareSize >= 3) checkAndClearOwner(size, x, y, mintedLand, index, quadCompareSize);
+    }
+
+    function isQuadMinted(
+        Land[] memory mintedLand,
+        uint256 x,
+        uint256 y,
+        uint256 size,
+        uint256 index
+    ) internal pure returns (bool) {
+        for (uint256 i = 0; i <= index; i++) {
+            Land memory land = mintedLand[i];
+            if (land.size != 0 && land.size > size) {
```

Comment:
In which case land.size == 0  ?
The condition land.size == size doesn't make sense here ?